### DISCUSSION TITLE
What's the replacement for mapTemplate

### DISCUSSION THREAD

**Discussion by j** [ai/nanostores#180] [Q&A] [answered]
We were using `mapTemplate` and are now wondering how to get ready to migrate.

**Comment by ai** [comment, marked as answer]
Use something like this:

```js
export function getPost (id) {
  if (!Template.cache[id]) {
      let store = atom()
      // add all callbacks
      Template.cache[id] = store
    }
    return Template.cache[id]
}

getPost.cache = {}
```

**Reply by j** [reply]
I sort of was figuring this.  I didn't know if it was an anti pattern to create an "unknown" store on the fly. I've been tempted to instead create stores in a root context.

**Reply by ai** [reply]
It is OK to create stores in runtime.

We deprecated this helpers because it is easy to create it manually, but many new developers didn't understand how to use this helper.

**Reply by j** [reply]
I think the deprecation is fine IMO. To me it wasn't clear how it handled GC.  If I can manually create stores, I can also sort of control my cleanup if needed.

I look forward to that context based version too.
